src/main/java/seedu/address/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SORT;
+
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.SortField;
+
+
+/**
+ * Parses input arguments and creates a new ListCommand object
+ */
+public class ListCommandParser implements Parser<ListCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns a {@code ListCommand} for execution.
+     *
+     * @throws ParseException if the user input does not conform to the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+
+        SortField sortField;
+        ArgumentMultimap argMultiMap = ArgumentTokenizer.tokenize(args, PREFIX_SORT);
+
+        // If sort field is missing
+        if (!argMultiMap.getValue(PREFIX_SORT).isPresent()) {
```

Comment:
Currently the command `list balabala` lists persons by no field, maybe it's better to throw an incorrect command format exception for this in the parse function?

Reply:
Yeah I thought about doing that, but I realise a `ParseException` is used to indicate an error. However in the case of `list balabala`, even though `balabala` is invalid, the overall command is still valid because we still want to show the list but without any sorting performed. That's why I chose not to throw an exception since we still want to continue execution.

Also the method to tokenize the input, `ArgumentTokenizer.tokenize()`, won't parse user input which doesn't have a prefix. So I wouldn't be able to differentiate `list` and `list balabala` using the `tokenize()` method. In both cases, the `PREFIX_SORT` won't exist in the `ArgumentMultimap`.

But I see where you're coming from in that erroneous input should be stopped even if we are able to continue execution. Thanks for raising this up!